src/test/java/com/urswolfer/gerrit/client/rest/http/groups/GroupsRestClientTest.java

```diff
@@ -231,4 +233,138 @@ public TestListRequest withStart(int start) {
+    	            
+    	        ), new Function<GroupQueryTestCase, GroupQueryTestCase[]>() {
+    	            @Override
+    	            public GroupQueryTestCase[] apply(GroupQueryTestCase testCase) {
+    	                return new GroupQueryTestCase[]{testCase};
+    	            }
+    	        }).iterator();
+    }
+    private static GroupQueryTestCase queryTestCase() {
+        return new GroupQueryTestCase();
+    }
+    private static final class GroupQueryTestCase {
+        private TestQueryRequest queryParameter = new TestQueryRequest();
+        private String expectedUrl;
+        private JsonElement mockJsonElement = EasyMock.createMock(JsonElement.class);
+        private GerritRestClient gerritRestClient;
+        private GroupsParser groupsParser;
+
+        public GroupQueryTestCase withQueryParameter(TestQueryRequest listParameter) {
+            this.queryParameter = listParameter;
+            return this;
+        }
+
+        public GroupQueryTestCase expectUrl(String expectedUrl) {
+            this.expectedUrl = expectedUrl;
+            return this;
+        }
+
+        public GroupQueryTestCase execute() throws Exception {
+            GroupsRestClient groupsRestClient = getGroupsRestClient();
+            Groups.QueryRequest query = groupsRestClient.query();
+            queryParameter.apply(query).get();
+            return this;
+        }
+
+        public void verify() {
+            EasyMock.verify(gerritRestClient, groupsParser);
+        }
+
+        public GroupsRestClient getGroupsRestClient() throws Exception {
+            return new GroupsRestClient(
+                setupGerritRestClient(),
+                setupGroupsParser()
+            );
+        }
+
+        public GerritRestClient setupGerritRestClient() throws Exception {
+            gerritRestClient = EasyMock.createMock(GerritRestClient.class);
+            EasyMock.expect(gerritRestClient.getRequest(expectedUrl))
+                .andReturn(mockJsonElement)
+                .once();
+            EasyMock.replay(gerritRestClient);
+            return gerritRestClient;
+        }
+
+        public GroupsParser setupGroupsParser() throws Exception {
+            groupsParser = EasyMock.createMock(GroupsParser.class);
+            EasyMock.expect(groupsParser.parseGroupInfos(mockJsonElement))
+                .andReturn(new ArrayList<GroupInfo>())
+                .once();
+            EasyMock.replay(groupsParser);
+            return groupsParser;
+        }
+
+        @Override
+        public String toString() {
+            return expectedUrl;
+        }
+    }
+    private final static class TestQueryRequest {
+        private EnumSet<ListGroupsOption> options =
+            EnumSet.noneOf(ListGroupsOption.class);
+        private List<String> projects = new ArrayList<String>();
+        private List<String> groups = new ArrayList<String>();
+
+       
```

Comment:
Remove this empty line (and some of the following where there is more than one).